core/plugins/org.fusesource.ide.camel.model.service.v2_17_0_redhat_630254/src/org/fusesource/ide/camel/model/service/internal/CamelService.java

```diff
@@ -0,0 +1,200 @@
+	}	
+	
+	/* (non-Javadoc)
+	 * @see org.fusesource.ide.camel.model.service.core.ICamelManagerService#createEndpointUri(java.lang.String, java.util.Map)
+	 */
+	@Override
+	public String createEndpointUri(String scheme, Map<String, String> properties) throws URISyntaxException {
+		if (catalog == null) catalog = new DefaultCamelCatalog();
+		return catalog.asEndpointUri(scheme, properties, ENCODE_DEFAULT);
+	}
+	
+	/* (non-Javadoc)
+	 * @see org.fusesource.ide.camel.model.service.core.ICamelManagerService#createEndpointUri(java.lang.String, java.util.Map, boolean)
+	 */
+	@Override
+	public String createEndpointUri(String scheme, Map<String, String> properties, boolean encode)
+			throws URISyntaxException {
+		if (catalog == null) catalog = new DefaultCamelCatalog();
+		return catalog.asEndpointUri(scheme, properties, encode);
+	}
+	
+	/* (non-Javadoc)
+	 * @see org.fusesource.ide.camel.model.service.core.ICamelManagerService#getEndpointProperties(java.lang.String)
+	 */
+	@Override
+	public Map<String, String> getEndpointProperties(String uri) throws URISyntaxException {
+		if (catalog == null) catalog = new DefaultCamelCatalog();
+		return catalog.endpointProperties(uri);
+	}
+	
+	/* (non-Javadoc)
+	 * @see org.fusesource.ide.camel.model.service.core.ICamelManagerService#createEndpointXml(java.lang.String, java.util.Map)
+	 */
+	@Override
+	public String createEndpointXml(String scheme, Map<String, String> properties) throws URISyntaxException {
+		if (catalog == null) catalog = new DefaultCamelCatalog();
+		return catalog.asEndpointUriXml(scheme, properties, ENCODE_DEFAULT);
+	}
+
+	/* (non-Javadoc)
+	 * @see org.fusesource.ide.camel.model.service.core.ICamelManagerService#createEndpointXml(java.lang.String, java.util.Map, boolean)
+	 */
+	@Override
+	public String createEndpointXml(String scheme, Map<String, String> properties, boolean encode)
+			throws URISyntaxException {
+		if (catalog == null) catalog = new DefaultCamelCatalog();
+		return catalog.asEndpointUriXml(scheme, properties, encode);
+	}
+	
+	/* (non-Javadoc)
+	 * @see org.fusesource.ide.camel.model.service.core.ICamelManagerService#getEndpointScheme(java.lang.String)
+	 */
+	@Override
+	public String getEndpointScheme(String uri) {
+		if (catalog == null) catalog = new DefaultCamelCatalog();
+		return catalog.endpointComponentName(uri);
+	}
+	
+	/* (non-Javadoc)
+	 * @see org.fusesource.ide.camel.model.service.core.ICamelManagerService#testExpression(java.lang.String, java.lang.String)
+	 */
+	@Override
+	public String testExpression(String language, String expression) {
+		String result = null;
+		org.apache.camel.impl.DefaultCamelContext ctx = new org.apache.camel.impl.DefaultCamelContext();
+		try {
+			ctx.resolveLanguage(language).createPredicate(expression.replaceAll("\n", "").replaceAll("\r", "").trim());
+			result = null;
+		} catch (Exception ex) {
+			result = ex.getMessage();
+		} finally {
+			try {
+				ctx.shutdown();
+			} catch (Exception ex) {
```

Comment:
would like to avoid changing behavior on this old duplicated code which will be removed for 10.0